### PR TITLE
Make mbedtls build

### PR DESCRIPTION
**Pull request by npmccallum**
This is a fork of openssl-src. It provides utilities for building
mbedtls into a rust application. An mbedtls-sys crate will also need to
be produced.

This crate is not yet complete. We currently don't support any
configuration options. These need to be added. We also need to test
against multiple architectures and libc implementations. There is
currently no Windows support.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +17,13 @@ at your option.
...
-for inclusion in Serde by you, as defined in the Apache-2.0 license, shall be
-dual licensed as above, without any additional terms or conditions.
+Unless you explicitly state otherwise, any contribution intentionally
+submitted for inclusion in mbedtls-src by you, as defined in the
+Apache-2.0 license, shall be dual licensed as above, without any
+additional terms or conditions.
+
+### Foundations
+
+This crate is heavily based on the work of Alex Crighton's crate for [OpenSSL].
```
Comment (jethrogb) [inline README.md]: Typo

Comment (npmccallum) [reply README.md]: I'm probably blind. What am I missing?

Comment (jethrogb) [reply README.md]: Alex's name

Comment (npmccallum) [reply README.md]: Fixed.

**Comment by npmccallum** [comment]
@jethrogb Are you still wanting to do another round of review?

**Comment by jethrogb** [comment]
I haven't looked at it in detail, but since all the stuff that makes building MbedTLS complicated (i.e. features/config.h) isn't handled in this PR, this seems fine.

**Comment by npmccallum** [comment]
@jethrogb Exactly. We'll handle this in future pull requests. Right now we just build with the defaults. I'm going to merge.

**Pull request merged**